doc/dev/osd_internals/manifest.rst

```diff
@@ -46,24 +48,64 @@ There are some rough edges we may want to improve:
     do_osd_ops.  This probably isn't desirable, see Snapshots section
     below for some options on how generally to mix these operations
     with snapshots.
+    --> Modify SET_CHUNK to set user_modify = false not to trigger a new clone.
   * Appears to assume that the corresponding section of the object
     does not exist (sets FLAG_MISSING) but does not check whether the
     corresponding extent exists already in the object.
+    --> Set FLAG_DIRTY if the corresponding section of the object exists.
+
   * Appears to clear the manifest unconditionally if not chunked,
     that's probably wrong.  We should return an error if it's a
     REDIRECT.
+    --> add the following lines ::
+
+	case CEPH_OSD_OP_SET_CHUNK:
+	  if (oi.manifest.is_redirect()) {
+	    result = -EINVAL;
+	    goto fail;
+	  }
+
 
 * TIER_PROMOTE:
 
   * Document control flow in this file.  The CHUNKED case is
     particularly unintuitive and could use some explanation.
+    --> 
+    Based on the chunk_map in a head object (each chunk's state can be 
+    dirty or missing), chunked objects for the head object 
+    can be retrieved when tier_promote is invoked.
+    After promotion is completed, all chunks in the object should be clean state.
+
+    One thing to note is snapshotted manifest object.
+    Suppose that there are 10(9), 5(4). If we want to read clone id 5,
+    need to read chunks according to the chunk_map for clone id 5.
+    But, because the base tier keeps object_info_t in the case of manifest object,
+    the read with snapshot can work.
+    If clone id 5 is dirty or clean, the chunks will be copied from the base tier.
+    If clone id 5 is missing, the chunks will be copied from the lower tier 
```

Comment:
Is it possible for clone id 5 to be completely missing?  Or do you intend for clones to work like the head where at least the object_info and therefore manifest info area always present?

Reply:
I mean,  the manifest info area always present even if all clones are flushed. So, the missing state means that manifest info exists but the corresponding chunks can be missing in the base tier.

Reply:
Ok, so you mean *for each chunk* if the chunk is dirty or clean it will be read directly from the base tier, or copied from the lower tier otherwise.

Reply:
Right